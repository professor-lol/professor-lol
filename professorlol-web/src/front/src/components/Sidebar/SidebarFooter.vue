<template>
  <div v-if="hasSlotDefault" class="sidebar-footer">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'sidebar-footer',
  computed: {
    hasSlotDefault() {
      return !!this.$slots.default;
    }
  }
}
</script>
