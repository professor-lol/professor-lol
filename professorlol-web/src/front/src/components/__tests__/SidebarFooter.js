import { mount } from '@vue/test-utils'
import SidebarFooter from "../Sidebar/SidebarFooter";

describe("SidebarFooter.vue", () => {
  // Inspect the raw component options
  it('has hasSlotDefault computed property', () => {
    expect(typeof SidebarFooter.computed.hasSlotDefault).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(SidebarFooter, { slots: { default: 'test'}})
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.classes()).toContain('sidebar-footer')
  })
});
