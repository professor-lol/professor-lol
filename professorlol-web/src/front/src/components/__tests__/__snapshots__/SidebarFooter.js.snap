// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarFooter.vue renders correctly 1`] = `
<div
  class="sidebar-footer"
>
  test
</div>
`;
